src/transaction.cpp

```diff
@@ -1083,10 +1084,15 @@ bool wsrep::transaction::total_order_bf_abort(
     wsrep::unique_lock<wsrep::mutex>& lock WSREP_UNUSED,
     wsrep::seqno bf_seqno)
 {
+    /* We must set this flag before entering bf_abort() in order
+     * to streaming_rollback() work correctly. The flag will be
+     * unset if BF abort was not allowed. Note that we rely in
+     * bf_abort() not to release lock if the BF abort is not allowed. */
+    bf_aborted_in_total_order_ = true;
```

Comment:
Ahhh... I always wondered why bf_aborted_in_total_order_ was set only after bf abort, but never investigated deeply enough to realize it was a bug...
Thanks for fixing this!